tests/locking.html

```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<script src="resources/text-encode-transform.js"></script>
+<script src="resources/transform-stream.js"></script>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+'use strict';
+
+const string = 'I \u{1F499} streams';
+const bytes = [73, 32, 240, 159, 146, 153, 32, 115, 116, 114, 101,
+                             97, 109, 115];
+const bytesAsArray = new Uint8Array(bytes);
+
+test(() => {
+  const encoder = new TextEncoder();
+  const writer = encoder.writable.getWriter();
+  assert_throws(new TypeError(), () => encoder.encode(string),
+                'encode should throw');
+  writer.releaseLock();
+  assert_array_equals(encoder.encode(string), bytes, 'encode should not throw');
```

Comment:
ah, I see. It just works for both arrays and typed arrays.

Reply:
Yes, I'm not sure why assert_array_equals isn't type strict when all the other asserts are, but it works well here.